python/lvmdrp/utils/metadata.py

```diff
@@ -0,0 +1,828 @@
+    if stage is not None:
+        logger.info(f"filtering by {stage = }")
+        query.append("stage == @stage")
+    if status is not None:
+        logger.info(f"filtering by {status = }")
+        query.append("status == @status")
+
+    if query:
+        query = " and ".join(query)
+        metadata.query(query, inplace=True)
+
+    return metadata
+
+
+def _create_store(observatory):
+    """return the metadata store given a path
+
+    Parameters
+    ----------
+    observatory: str
+        observatory for which a metadata store will be created
+    """
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    logger.info(f"creating metadata store '{metadata_path}'")
+    store = h5py.File(metadata_path, mode="w")
+
+    # add observatory group if needed
+    if observatory not in store:
+        store.create_group(f"{observatory}/raw")
+        store.create_group(f"{observatory}/master")
+
+    store.close()
+
+
+def _del_store():
+    """delete the entire HDF store if exists"""
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    if os.path.isfile(metadata_path):
+        logger.info(f"removing metadata store '{metadata_path}'")
+        os.remove(metadata_path)
+    else:
+        logger.warning(f"no '{metadata_path}' store found")
+
+
+def _load_store(observatory, mode="r"):
+    """return the metadata store given a observatory
+
+    Parameters
+    ----------
+    observatory : str
+        metadata observatory from which a store will be loaded
+    mode : str, optional
+        the mode in which to open the HDF5 store, by default 'r'
+
+    Returns
+    -------
+    h5py.Group
+        the metadata store for the given observatory
+    """
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    logger.info(f"loading metadata store from '{metadata_path}'")
+    store = h5py.File(metadata_path, mode=mode)
+
+    return store[observatory]
+
+
+def extract_metadata(mjd, frames_paths):
+    """return dataframe with metadata extracted from given frames list
+
+    this function will extract metadata from FITS headers given a list of
+    frames.
+
+    Parameters
+    ----------
+    mjd : int
+        MJD from which the metadata will be extracted
+    frames_paths : list_like
+        list of frames paths in the local mirror of SAS
+
+    Returns
+    -------
+    pandas.DataFrame
+        dataframe containing the extracted metadata
+    """
+    new_metadata = {}
+    # extract metadata
+    nframes = len(frames_paths)
+    logger.info(f"extracting metadata from {nframes} frames")
+    iterator = tqdm(
+        enumerate(frames_paths),
+        total=nframes,
+        desc=f"extracting metadata from MJD = {mjd}",
+        ascii=True,
+        unit="frame",
+    )
+    for i, frame_path in iterator:
+        header = fits.getheader(frame_path, ext=0)
+        new_metadata[i] = [
+            "n" if header.get("OBSERVAT") != "LCO" else "s",
+            header.get("IMAGETYP"),
```

Comment:
The `TILEID` header keyword should be included here.   If not already in there, can you also include the spectrograph ID and the file name as columns?  Just the file name is fine, not the full path.

Reply:
Ok sounds good.